Add test instructions and some tidying

samuelallan72 opened the pull request:
## Description

Add instructions for manually testing the plugin on a devstack.

Also do some tidying of the repo, removing unused files, clarifying the license, and fixing some minor issues I discovered on the way.

Also add a pull request template to remind developers to do the manual testing, because there are no tests in CI (apart from the quality lints).

## Supporting information

Private-ref: [BB-10035](https://tasks.opencraft.com/browse/BB-10035)

## Test instructions

- View the rendered README.md and ./docs/manual-test-plan.md to check for formatting issues.
- Follow the test instructions and devstack installation instructions in the README and verify everything works as expected.
- Follow the playbook in ./docs/manual-test-plan.md and verify the instructions are clear, easy to follow, and work as expected.

## Deadline

None

samuelallan72 committed:
test: add manual test instructions, tidy repo

- remove unused files
- update dependencies
- do some minor tidying
- add manual test instructions
- clarify AGPLv3 license

Private-ref: https://tasks.opencraft.com/browse/BB-10035

Kelketek commented on README.md:
```diff
@@ -17,10 +17,53 @@ Updated for standalone, Juniper/Py3 by @bryanlandia
+
+# This directory won't be autodetected for build-time mount by Tutor,
+# so we need to manually configure Tutor for it.
+mkdir -p "$TUTOR_PLUGINS_ROOT"
```
This variable is undefined. Maybe there should be a line before this one?

```
TUTOR_PLUGINS_ROOT=$(tutor plugins printroot)
```

Kelketek commented on README.md:
````diff
@@ -32,9 +75,22 @@ In the LMS shell, cd to the plugin directory:
+Then you can run the migrations test:
+
+```
+make test_migrations
````
```console
fox@rebel$ tutor dev exec lms -- bash
docker compose -f /home/fox/.local/share/tutor/env/local/docker-compose.yml -f /home/fox/.local/share/tutor/env/dev/docker-compose.yml --project-name tutor_dev exec lms bash
app@5167f59bad7b:~/edx-platform$ cd /mnt/legacy-appsembler-api
app@5167f59bad7b:/mnt/legacy-appsembler-api$ make test_migrations
tox -e migrations
migrations: install_deps> python -I -m pip install -r /mnt/legacy-appsembler-api/requirements/base.txt
.pkg: install_requires> python -I -m pip install 'setuptools>=40.8.0' wheel
.pkg: _optional_hooks> python /openedx/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist> python /openedx/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_wheel> python /openedx/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel> python /openedx/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist> python /openedx/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
migrations: install_package_deps> python -I -m pip install Django djangorestframework
migrations: install_package> python -I -m pip install --force-reinstall --no-deps /mnt/legacy-appsembler-api/.tox/.tmp/package/1/shoppingcart-0.1.0.tar.gz
migrations: commands[0]> python ./manage.py makemigrations shoppingcart --check --dry-run --verbosity 3
Traceback (most recent call last):
  File "/mnt/legacy-appsembler-api/./manage.py", line 29, in <module>
    execute_from_command_line(sys.argv)
  File "/mnt/legacy-appsembler-api/.tox/migrations/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/mnt/legacy-appsembler-api/.tox/migrations/lib/python3.11/site-packages/django/core/management/__init__.py", line 416, in execute
    django.setup()
  File "/mnt/legacy-appsembler-api/.tox/migrations/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/mnt/legacy-appsembler-api/.tox/migrations/lib/python3.11/site-packages/django/apps/registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/legacy-appsembler-api/.tox/migrations/lib/python3.11/site-packages/django/apps/config.py", line 123, in create
    mod = import_module(mod_path)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/pyenv/versions/3.11.8/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/mnt/legacy-appsembler-api/shoppingcart/apps.py", line 6, in <module>
    from openedx.core.djangoapps.plugins.constants import PluginURLs, ProjectType
ModuleNotFoundError: No module named 'openedx'
migrations: exit 1 (0.25 seconds) /mnt/legacy-appsembler-api> python ./manage.py makemigrations shoppingcart --check --dry-run --verbosity 3 pid=147
  migrations: FAIL code 1 (9.95=setup[9.70]+cmd[0.25] seconds)
  evaluation failed :( (10.05 seconds)
make: *** [Makefile:67: test_migrations] Error 1

samuelallan72 replied:
Ah sorry, you must have had an older revision here, this is fixed already (I moved the command from Tox to directly in the Makefile, so it's not in the Tox venv any longer). :sweat_smile: 

I was force pushing to the branch while it was still in draft mode, so that might have made things confusing. That's probably a bad habit sorry.

Kelketek replied:
Ah, you're right, I was on the wrong branch. I'll try again!

Kelketek commented on docs/manual-test-plan.md:
````diff
@@ -0,0 +1,809 @@
+Test it by logging in (requires the CSRF token cookie AND the CSRF token header):
+
+```sh
+curl --show-headers -X POST 'http://local.openedx.io:8000/appsembler_api/v0/account/login_session/' \
````
`--show-headers` isn't a flag curl seems to recognize. Not sure if it's a version issue. `--include` seems to be the most proximate to what you're asking for here, and `-v` should also work, though it's spammier.

And it appears they **DO** need to be shown, as there's a missing piece of info that should be after this segment-- one noting that the CSRF token will be updated and the user needs to change the environment variable to the new one.

samuelallan72 replied:
Oh that changes the csrf token does it? Hmm ok, let me run through this in order myself and see what's happening. :thinking: 

`--show-headers` works for me, so it must be a version thing. I'll use something else so others don't run into this issue too though.

For reference:

```console
$ curl --version
curl 8.16.0 (x86_64-alpine-linux-musl) libcurl/8.16.0 OpenSSL/3.5.3 zlib/1.3.1 brotli/1.1.0 zstd/1.5.7 c-ares/1.34.5 libidn2/2.3.8 libpsl/0.21.5 nghttp2/1.67.1 nghttp3/1.11.0
Release-Date: 2025-09-10
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTP3 HTTPS-proxy IDN IPv6 Largefile libz NTLM PSL SSL threadsafe TLS-SRP UnixSockets zstd

$ curl --show-headers 'https://example.com'
HTTP/2 200
accept-ranges: bytes
content-type: text/html
...

samuelallan72 replied:
Ah I see, that makes it difficult. :/  I think I'll add a note to re-run the csrf token step, to avoid needing to go through headers and such.

samuelallan72 replied:
Annoyingly that login_session endpoint requires the CSRF cookie _and_ the CSRF header.

Kelketek commented on docs/manual-test-plan.md:
````diff
@@ -0,0 +1,809 @@
+
+```sh
+curl -X POST 'http://local.openedx.io:8000/appsembler_api/v0/account/login_session/' \
+  --header 'Content-Type: multipart/form-data' \
````
`--cookie-jar` option missing here. Otherwise this fails for lack of CSRF cookie.

samuelallan72 replied:
Oops, I wrote cookie instead of cookie jar. Yeah I need to run through these again - this was some last minute hacking after discovering it depended on cookies. I've been primarily testing in Yaak, which automatically tracks cookies, which is helpful but makes it harder to see if an exported curl command will work or not...

samuelallan72 replied:
Wait, no this also needs `--cookie`.

Kelketek commented on docs/manual-test-plan.md:
```diff
@@ -0,0 +1,809 @@
+curl -X POST 'http://local.openedx.io:8000/appsembler_api/v0/enrollment-codes/status' \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "action": "restore",
```
This is the restoration command rather than the enrollment command. Probably copy/pasted from the wrong section.

samuelallan72 replied:
Yep that's a bad copy/paste, thanks.

Kelketek commented on docs/manual-test-plan.md:
````diff
@@ -0,0 +1,809 @@
+
+Verify the response has status 200 and the body has a json list of all users:
+
+```json
````
```suggestion
```json5
```

Kelketek commented on docs/manual-test-plan.md:
```diff
@@ -0,0 +1,809 @@
+    "is_active": true,
+    "date_joined": "2025-09-29T07:05:03.092197Z"
+  },
+  ...
```
```suggestion
  // ...
]
```

Kelketek replied:
The renderer on GitHub does not like the ellipsis here:

<img width="697" height="253" alt="image" src="https://github.com/user-attachments/assets/33c7319e-84ef-4e60-8900-be2008cfac2f" />

And on the diff from the review page:

<img width="697" height="253" alt="image" src="https://github.com/user-attachments/assets/5d4caa1e-aa15-415b-9b56-1fcf69f7ebdb" />

samuelallan72 replied:
Ah that really messes up the diff highlighting, wow. I was hoping for the simple red ellipsis to show it's been truncated. I like your json5 method, thanks. :+1: :+1:

Kelketek reviewed (changes requested):
@samuelallan72 I went ahead and reviewed this since I knew @tecoholic would already have a setup with it working, but I wouldn't, so it would be a better test. I did find a few issues.

@tecoholic Since @samuelallan72 is out the rest of the week, if you happen to have free time (other tasks in this epic are higher priority) and can address the notes, feel free to do so and I'll do another review pass.

samuelallan72 commented:
@Kelketek @tecoholic 

> Since @samuelallan72 is out the rest of the week, 

I'll be back tomorrow (Thursday); I just took today off. :smile:  Of course if you have time @tecoholic , feel free to push any changes! :)

Kelketek commented:
@samuelallan72 Whoops! I misread your message on Mattermost. Cool-- noted, and thanks!

samuelallan72 committed:
Address review, fix issues following manual tests

samuelallan72 committed:
Tweak and improve usability of the manual tests

samuelallan72 commented:
@Kelketek I addressed your review comments and made some general usability improvements to the manual tests. This is ready for another review pass. :) cc @tecoholic

Kelketek reviewed (approved):
:+1:

- [x] I tested this: Migration tests and manual test plan
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~
- [x] Includes documentation
- [ ] ~~Added to the [Code Drift](https://github.com/orgs/open-craft/projects/1) project board (for backports)~~
